pytorch_lightning/trainer/training_loop.py

```diff
@@ -474,7 +474,13 @@ def run_training_epoch(self):
             # -----------------------------------------
             # SAVE METRICS TO LOGGERS
             # -----------------------------------------
-            self.trainer.logger_connector.save_train_loop_metrics_to_loggers(batch_idx, batch_output)
+            should_log_metrics = ((batch_idx + 1) % self.trainer.row_log_interval == 0
```

Comment:
The logic should be moved back to the `save_train_loop_metrics_to_loggers` function to keep the training loop as clean as possible.

Reply:
Done.